src/modules/ens/sagas.ts

```diff
@@ -497,6 +501,52 @@ export function* ensSaga(builderClient: BuilderClient, ensApi: ENSApi, worldsAPI
     }
   }
 
+  function* handleFetchContributableNamesRequest() {
+    try {
+      let names: ContributableDomain[] = yield call([ensApi, worldsAPIContent.fetchContributableDomains])
+
+      const bannedNames: string[] = yield call([lists, 'fetchBannedNames'])
+      const bannedNamesSet = new Set(bannedNames.map(x => x.toLowerCase()))
+
+      names = names.filter(({ name }) => name.split('.').every(nameSegment => !bannedNamesSet.has(nameSegment)))
+      const ensDomains: string[] = []
+      const nameDomains: string[] = []
+      names.forEach(({ name }) => {
+        if (name.includes('.dcl.eth')) {
+          nameDomains.push(name.replace('.dcl.eth', ''))
+        } else {
+          ensDomains.push(name)
+        }
+      })
+      const ownerByNameDomain: Record<string, string> = yield call([marketplace, 'fetchENSOwnerByDomain'], nameDomains)
+      const ownerByEnsDomain: Record<string, string> = yield call([ensApi, 'fetchExternalENSOwners'], ensDomains)
```

Comment:
What do you think about grouping these two calls in an `all` to reduce the load time?